### ISSUE TITLE
Fix S2187 FP: Support test attributes deriving from `ITestBuilder` in NUnit

### ISSUE THREAD

**Issue by dammejed** [SonarSource/sonar-dotnet#5564]
### Description

I'm seeing S2187 reported when using test fixtures that have test methods that derive from `AutoData` (an attribute defined in Autofixture.NUnit3) instead of the built-in `Test`, `TestMethod`, etc attributes from NUnit.

Such attributes implement the [`ITestBuilder`](https://docs.nunit.org/articles/nunit/extending-nunit/ITestBuilder-Interface.html) interface from NUnit, and these tests are correctly detected and run by visual studio.

### Repro steps

Create a test fixture in a correctly-configured Test project that uses the NUnit, AutoFixture, and Autofixture.NUnit3 packages, with the following code:

```csharp
using AutoFixture.NUnit3;
using NUnit.Framework;

namespace FalsePositive
{
    [TestFixture]
    public class FalsePositiveS2187TestFixture
    {
        [AutoData]
        public void AutoDataTest(string sut)
        {
            Assert.NotNull(sut);
        }
    }
}
```

### Expected behavior

No S2187 is raised.

### Actual behavior

S2187 is reported on the `FalsePositiveS2187TestFixture` class.

### Known workarounds

* Suppress the issue.
* Put a superfluous `Test` attribute on at least one method in the affected class.

### Related information

* Visual Studio 17.0.5
* SonarLint for VS 2022 version 6.1.0.45943
* I don't know what version of the C# plugin on SonarQube, but we have SonarQube 9.4.0.

As far as I understand it, this could be fixed by additionally blessing methods with attributes that implement the `ITestBuilder` interface, rather than the specific ones provided by NUnit.
That seems like it'd be fairly straightforward-- I'd be happy to investigate doing so myself.
